CIPs/0022.md

```diff
@@ -0,0 +1,40 @@
+---
+cip: 0022
+title: Standard for Injacted web3 Provider
+author: @daoauth @NAKsir-melody
+discussions-to: https://github.com/celo-org/celo-proposals/issues/74
+status: Draft
+type:
+category:
+created: 2020-10-29
+license: Apache 2.0
+---
+
+## Simple Summary
+Standard to find out the type of web3 provider to support dapps running in web browsers
+
+## Abstract
+- In the case of web DApps, you can only use [DSRV Celo Extension Wallet](https://chrome.google.com/webstore/detail/celo-desktop-wallet/kkilomkmpmkbdnfelcpgckmpcaemjcdh) or MetaMask Extension wallet, but a standard is required to distinguish them.
+- Through these standards, each provider can be identified and transactions can be sent to the Celo blockchain through appropriate operations.
+
+## Motivation
+To increase usability and accessibility by allowing celo DApps to be used not only in mobile environment but also in desktop environment.
+
+## Specification
+- Ready `ethereum.isMetaMask` property for MetaMask. Through this property, it can be seen that it is a provider injected in [MetaMask](https://docs.metamask.io/guide/ethereum-provider.html#properties).
+- So, it is suggested that providers injected in Celo have a property with the syntax of `celo[providerName]`.
+  * eg. Some injected provider has `celo.isDesktop` property. It's injected by DSRV Celo Extension wallet
+  * eg. For DApp browser has `celo.isMobile` too
+
+## Test Cases
+
```

Comment:
What are test cases, if any, needed to be specified to ensure this proposal is correct? Do curl commands suffice?

Reply:
It was suggested to add a parameter to find out what kind of provider is injected. For example, in the case of MetaMask, the MetaMask's definition is already done through a isMataMask parameter. Likewise, isDesktop means a provider injected by the DSRV wallet. And we can use the functions of MetaMask or DSRV wallet through web3.

Reply:
@YazzyYaz these aren't new RPC endpoints, rather they're properties that providers can expose to help dapps use the provider correctly. 

@daoauth does a corresponding EIP exist? Can we expand the specification to list suggested properties for injected providers?